es6/lib/createRouteHandler.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+// jshint -W040
+
+import Boom from 'boom';
+import camelCase from 'lodash/string/camelCase';
+import first from 'lodash/array/first';
+import forEach from 'lodash/collection/forEach';
+import isEmpty from 'lodash/lang/isEmpty';
+import mapKeys from 'lodash/object/mapKeys';
+import omit from 'lodash/object/omit';
+import rearg from 'lodash/function/rearg';
+import snakeCase from 'lodash/string/snakeCase';
+
+function apiController(options, knexClient) {
+  return {
+
+    create(request, reply) {
+      var resource;
+
+      if (options.tableHeaderFormat === 'snakeCase') {
+        var snakeCaseKeys = rearg(snakeCase, [1, 0]);
+        resource = mapKeys(request.payload, snakeCaseKeys);
+      } else {
+        resource = request.payload;
+      }
+
+      function createApiResponse(response) {
+        let apiResponse = { };
+        apiResponse[camelCase(options.index)] = first(response);
```

Comment:
ES6 can do computed properties in object literals, so if you like, this could be:

Feel free to leave it the way it is as well, just a new feature to be aware of.

```javascript
return {
  [camelCase(options.index)]: first(response)
};
```